sql: block FKs to/from TTL tables

otan opened the pull request:
This table blocks FKs to and from TTL tables. Note the RFC previously
only said to, but I actually thought of a new case and decided that is
was probably prudent to block this both ways. RFC is reflected
accordingly.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76409)
<!-- Reviewable:end -->

rafiss reviewed (approved):
lgtm!

though similar to my comment on the other PR, have we already considered blocking computed columns and partial index defs? (this would be for a future PR)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/76409#-:-Mvz2ztH8x0iAnZXo1Mh:ba9uymu)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)

___
*[-- commits, line 5 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/76409#-Mvz2c-E4KcBmgonmy65):*
nit: missed phrasing after "only said to"


<!-- Sent from Reviewable.io -->

otan committed:
sql: block FKs to/from TTL tables

This table blocks FKs to and from TTL tables. Note the RFC previously
only said to not block from TTL tables, but I actually thought of a
new case and decided that is was probably prudent to block this both
ways. RFC is reflected accordingly.

Release note: None

otan commented:
> have we already considered blocking computed columns and partial index defs?

i think we're ok here since:
(a) dropColumn knows how to handle if these exist
(b) with strict filtering these can be ignored, similar to a normal table lookup

TFTR!
bors r+

rafiss commented:
i guess it's an even more general question - do computed columns and partial index defs already play nice with `DEFAULT`/`ON UPDATE` columns? i think we might have an issue for that somewhere actually..
